File: VB/TestMyTreeListForum/MyToolButtonPainter.vb
Imports System
Imports System.Collections.Generic
Imports DevExpress.XtraTreeList
Imports System.Drawing



Namespace DevExpress.MyControl
	Public Class MyToolButtonPainter
		Public Sub New()
		End Sub



		Public Overridable Sub Draw(ByVal e As MyToolButtonDrawEventArgs)
			Dim vi As MyToolButtonViewInfo = e.ViewInfo
			e.Appearence.FillRectangle(e.Cache, vi.Bounds)


			If vi.Button.ButtonType = MyButtonType.Link Then
				e.Cache.DrawImage(My.Resources.CommentIcon, vi.ImageBounds)
				Dim fs As FontStyle = e.Appearence.Font.Style
				If vi.UnderCursor Then
					fs = fs Or FontStyle.Underline
				End If
				Using f As New Font(e.Appearence.Font, fs)
					e.Cache.DrawString(vi.Button.Caption, e.Appearence.GetFont(), e.Appearence.GetForeBrush(e.Cache), vi.CaptionBounds)
				End Using
				Return
			End If

			Dim bm As Bitmap
			If vi.Button.ButtonType = MyButtonType.Cancel Then
				bm = (If(vi.UnderCursor, My.Resources.CancelButtonIconFocused, My.Resources.CancelButtonIconUnfocused))
			Else
				bm = (If(vi.UnderCursor, My.Resources.CommitButtonIconFocused, My.Resources.CommitButtonIconUnfocused))
			End If

			e.Cache.DrawImage(bm, vi.ImageBounds)
		End Sub
	End Class
End Namespace

File: CS/TestMyTreeListForum/MyToolButtonPainter.cs
using System;
using System.Collections.Generic;
using DevExpress.XtraTreeList;
using System.Drawing;



namespace DevExpress.MyControl
{
    public class MyToolButtonPainter
    {
        public MyToolButtonPainter()
        {
        }



        public virtual void Draw(MyToolButtonDrawEventArgs e)
        {
            MyToolButtonViewInfo vi = e.ViewInfo;
            e.Appearence.FillRectangle(e.Cache, vi.Bounds);


            if (vi.Button.ButtonType == MyButtonType.Link)
            {                
                e.Cache.DrawImage(TestMyTreeListForum.Properties.Resources.CommentIcon, vi.ImageBounds);
                FontStyle fs = e.Appearence.Font.Style;
                if (vi.UnderCursor) fs |= FontStyle.Underline;
                using (Font f = new Font(e.Appearence.Font, fs))
                {             
                    e.Cache.DrawString(vi.Button.Caption, e.Appearence.GetFont(), e.Appearence.GetForeBrush(e.Cache), vi.CaptionBounds);
                }
                return;
            }

            Bitmap bm; 
            if (vi.Button.ButtonType == MyButtonType.Cancel)
                bm = (vi.UnderCursor ? TestMyTreeListForum.Properties.Resources.CancelButtonIconFocused 
                                     : TestMyTreeListForum.Properties.Resources.CancelButtonIconUnfocused);
            else
                bm = (vi.UnderCursor ? TestMyTreeListForum.Properties.Resources.CommitButtonIconFocused
                                     : TestMyTreeListForum.Properties.Resources.CommitButtonIconUnfocused);

            e.Cache.DrawImage(bm, vi.ImageBounds);
        }
    }
}
